docs/developer/architecture/metalk8s-rbac.rst

```diff
@@ -0,0 +1,112 @@
+RBAC for MetalK8s
+=================
+
+Context
+-------
+
+In order to secure the APIs of MetalK8s it makes sense to put in place a
+predefined RBAC strategy. To ease the operation of MetalK8s this RBAC strategy
+will be predefined.
+
+MetalK8s authentication for normal users is delegated to DEX OIDC plugin.
+
+MetalK8s APIs:
+
+- k8s
+- salt
+- prometheus
+- alertmanager
+- loki
+
+MetalK8s namespaces:
+
+- kube-system
+- metalk8s-auth
+- metalk8s-ingress
+- metalk8s-monitoring
+- metalk8s-solutions
+- metalk8s-ui
+
+MetalK8s Roles:
+
+- Cluster Admin Role
+- Platform Admin Role
+- Solution Admin Role
+- Platform Viewer Role
+
+MetalK8s Activities per Roles:
+
++----------------------------------+---------+----------+----------+----------+
+| Activities                       | Cluster | Platform | Solution | Platform |
+|                                  | Admin   | Admin    | Admin    | Viewer   |
++==================================+=========+==========+==========+==========+
+| Bootstrap Node administration    |   X     |          |          |          |
+| (backup, restore)                |         |          |          |          |
++----------------------------------+---------+----------+----------+----------+
+| Bind roles to users & groups     |   X     |    ?     |    ?     |          |
++----------------------------------+---------+----------+----------+----------+
+| Expansion of the cluster         |   X     |    X     |          |          |
+| cp,infra or wp node              |         |          |          |          |
```

Comment:
Not sure we need to mention that, to me, "Expansion of the cluster" is self explanatory.